engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/NadelTransformJavaCompat.kt

```diff
@@ -0,0 +1,112 @@
+package graphql.nadel.enginekt.transform
+
+import graphql.nadel.Service
+import graphql.nadel.ServiceExecutionResult
+import graphql.nadel.enginekt.NadelExecutionContext
+import graphql.nadel.enginekt.blueprint.NadelOverallExecutionBlueprint
+import graphql.nadel.enginekt.transform.query.NadelQueryTransformer
+import graphql.nadel.enginekt.transform.result.NadelResultInstruction
+import graphql.normalized.ExecutableNormalizedField
+import kotlinx.coroutines.future.asDeferred
+import java.util.concurrent.CompletableFuture
+
+/**
+ * See [NadelTransform]
+ */
+interface NadelTransformJavaCompat<State : Any> {
+    /**
+     * See [NadelTransform.isApplicable]
+     */
+    fun isApplicable(
+        executionContext: NadelExecutionContext,
+        executionBlueprint: NadelOverallExecutionBlueprint,
+        services: Map<String, Service>,
+        service: Service,
+        overallField: ExecutableNormalizedField,
+    ): CompletableFuture<State?>
+
+    /**
+     * See [NadelTransform.transformField]
+     */
+    fun transformField(
+        executionContext: NadelExecutionContext,
+        transformer: NadelQueryTransformer.Continuation,
+        executionBlueprint: NadelOverallExecutionBlueprint,
+        service: Service,
+        field: ExecutableNormalizedField,
+        state: State,
+    ): CompletableFuture<NadelTransformFieldResult>
+
+    /**
+     * See [NadelTransform.getResultInstructions]
+     */
+    fun getResultInstructions(
+        executionContext: NadelExecutionContext,
+        executionBlueprint: NadelOverallExecutionBlueprint,
+        service: Service,
+        overallField: ExecutableNormalizedField,
+        underlyingParentField: ExecutableNormalizedField?,
+        result: ServiceExecutionResult,
+        state: State,
+    ): CompletableFuture<List<NadelResultInstruction>>
+
+    companion object {
+        fun <State : Any> create(compat: NadelTransformJavaCompat<State>): NadelTransform<State> {
```

Comment:
We should probably add `@JvmStatic` to this function.

In Kotlin without that annotation there is no concept of real static, and for good reason. Effectively if you don't put that annotation on, in Java you have to call it like so:

```
NadelTransformJavaCompat.Companion.create(…)
```

vs if you have `@JvmStatic` on it

```
NadelTransformJavaCompat.create(…)
```